native/cocos/platform/android/java/vendor/play/google/play/TaskManager.java

```diff
@@ -0,0 +1,214 @@
+                }
+            });
+        });
+
+        return putTask(newTask);
+    }
+    static int addOnFailureListener(int taskId, int listenerId) {
+        if(!_taskMap.containsKey(taskId)) {
+            Log.e(TAG, "Called task id does not exist");
+            return -1;
+        }
+        Task<?> newTask = _taskMap.get(taskId).addOnFailureListener((var)->{
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    int nextExceptionId = ++_nextExceptionId;
+                    _exceptionsMap.put(nextExceptionId, var);
+                    onTaskFailureNative(taskId, listenerId, var, nextExceptionId);
+                }
+            });
+        });
+        return putTask(newTask);
+    }
+    static int addOnSuccessListener(int taskId, int listenerId) {
+        if(!_taskMap.containsKey(taskId)) {
+            Log.e(TAG, "Called task id does not exist");
+            return -1;
+        }
+        Task<?> newTask = _taskMap.get(taskId).addOnSuccessListener((var)->{
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    onTaskSuccessNative(taskId, listenerId, var);
+                }
+            });
+        });
+        return putTask(newTask);
+    }
+
+
+    private static <TResult, TContinuationResult> Task<TContinuationResult> continueWithTaskForAny(
+        Task<TResult> task, final Continuation<TResult, TContinuationResult> continuation) {
+
+        return task.continueWith(new Continuation<TResult, TContinuationResult>() {
+            @Override
+            public TContinuationResult then(Task<TResult> task) throws Exception {
+                return continuation.then(task);
+            }
+        });
+    }
+    static int continueWith(int taskId, int listenerId) {
+        if(!_taskMap.containsKey(taskId)) {
+            Log.e(TAG, "Called task id does not exist");
+            return -1;
+        }
+        Task<Object> task = (Task<Object>)_taskMap.get(taskId);
+        Task<?> newTask = task.continueWith(new Continuation<Object, Object>() {
+            @Override
+            public Object then(Task<Object> task) throws Exception {
+                final CountDownLatch latch = new CountDownLatch(1);
+                final AtomicReference<Object> resultHolder = new AtomicReference<>();
+                int calllbackTaksId = putTask(task);
+                if (task.isSuccessful()) {
+                    CocosHelper.runOnGameThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            int calllbackTaksId = putTask(task);
+                            resultHolder.set(onContinueWithNative(taskId, listenerId, calllbackTaksId));
+                            latch.countDown();
+                        }
+                    });
+                    latch.await();
```

Comment:
Which thread does latch.await live in ?

Reply:
The default task is the main thread running

Reply:
If so, the `await` may cause ANR issues. Could we custom the default task queue to sub-thread?

Reply:
yep,I switched to other threads for execution.https://github.com/cocos/cocos-engine/pull/18432